awscrt/common.py

```diff
@@ -18,3 +18,23 @@ def get_cpu_count_for_group(group_idx: int) -> int:
     Returns number of processors in a given group.
     """
     return _awscrt.get_cpu_count_for_group(group_idx)
+
+class SystemEnvironment:
+
+    def __init__(self):
+        self._env = _awscrt.load_system_environment()
+
+        if self.is_ec2_nitro_instance():
+            self._detected_instance_type = _awscrt.get_ec2_instance_type(self._env)
+        else:
+            self._detected_instance_type = None       
+
+    def is_ec2_nitro_instance(self) -> bool:    
+        return _awscrt.is_env_ec2(self._env)
```

Comment:
Why mention "nitro" here, but not in the underlying C function?

I liked the `is_running_on_ec2` name better, FWIW

Reply:
It's really a nitro check, and ec2 is implied as a result. 

I'll make sure nitro is put in there and fuse it with the suggestion you made